test/e2e/clusteradm/upgrade_test.go

```diff
@@ -12,28 +18,63 @@ var _ = ginkgo.Describe("test clusteradm upgrade clustermanager and Klusterlets"
 		ginkgo.By("reset e2e environment...")
 		err := e2e.ClearEnv()
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
-		err = e2e.ResetEnv()
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
 
 	var err error
 
 	ginkgo.It("run cluster manager upgrade version latest ", func() {
-		ginkgo.Skip("Upgrade is skipped due to flaky when destroying control plane. Need to revisit it after fix cleanup issue")
-		err = e2e.Clusteradm().Upgrade(
-			"clustermanager",
-			"--bundle-version", "latest",
+		ginkgo.By("init hub with service account")
+		err = e2e.Clusteradm().Init(
 			"--context", e2e.Cluster().Hub().Context(),
 		)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "clusteradm init error")
 
-		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "clusteradm upgrade error")
+		ginkgo.By("Check the version of operator and agent")
+		gomega.Eventually(func() error {
+			operator, err := kubeClient.AppsV1().Deployments("open-cluster-management").Get(context.TODO(), "cluster-manager", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if operator.Spec.Template.Spec.Containers[0].Image != "quay.io/open-cluster-management/registration-operator:"+version.GetDefaultBundleVersion() {
+				return fmt.Errorf("version of the operator is not correct, get %s", operator.Spec.Template.Spec.Containers[0].Image)
+			}
+			registration, err := kubeClient.AppsV1().Deployments("open-cluster-management-agent").Get(
+				context.TODO(), "cluster-manager-registration-controller", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if registration.Spec.Template.Spec.Containers[0].Image != "quay.io/open-cluster-management/registration:"+version.GetDefaultBundleVersion() {
+				return fmt.Errorf("version of the operator is not correct, get %s", operator.Spec.Template.Spec.Containers[0].Image)
+			}
+			return nil
+		}, 5*time.Second, 30*time.Second).Should(gomega.Succeed())
 
 		err = e2e.Clusteradm().Upgrade(
-			"klusterlet",
+			"clustermanager",
```

Comment:
so we only need to test upgrading the clustermanager, not klusterlet, right?

Reply:
we need to, will add another test for klusterlet